### PR TITLE
Docs: search #606

### LINKED ISSUE #606: Docs: search
# Search

- [x] インデックス作成。
- [x] 検索テスト。
- [x] 日本語検索に対応する。
- [x] UI作成。
- [x] 検索処理実装。
- [x] リンク実装。
- [x] Use code block design for colors
- [x] Remove markdown symbols
- [x] Keyboard controls

### PR DESCRIPTION
**Pull request by milkyskies**
closes #606

### CODE REVIEW

File: src/routes/docs/search_modale.svelte [hunk 1]
```diff
@@ -0,0 +1,241 @@
+<script lang="ts">
```
Comment (sakata0411) [inline src/routes/docs/search_modale.svelte:1]: - [ ] Rename file to search-modale.svelte

File: src/lib/docs/search_result_context.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
+import type Fuse from 'fuse.js'
+
+export interface SplitContextPortion {
+	text: string
+	is_match: boolean
+	string_index: number
```
Comment (sakata0411) [inline src/lib/docs/search_result_context.ts:6]: first_character_index

File: src/lib/docs/search_result_context.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
+import type Fuse from 'fuse.js'
+
+export interface SplitContextPortion {
```
Comment (milkyskies) [inline src/lib/docs/search_result_context.ts:3]: type

File: src/lib/docs/search.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+export interface Context {
+	full_context: string
+	split_context: {
+		before_match: string
+		matched_text: string
+		after_match: string
+	}
+}
+
```
Comment (milkyskies) [inline src/lib/docs/search.ts:18]: type

**Pull request merged**